dvc/dagascii.py

```diff
@@ -12,6 +15,42 @@
 from grandalf.routing import EdgeViewer
 from grandalf.routing import route_with_lines
 
+from dvc.env import DVC_PAGER
+from dvc.utils import boxify, is_exec_found
+
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_PAGER = "less"
+DEFAULT_PAGER_FORMATTED = "{} --chop-long-lines --clear-screen".format(
+    DEFAULT_PAGER
+)
+DVC_PAGER_INFO = """\
+Less command line tool is missing.
+Install & add less tool to your PATH env. var. to automatically send output \
+to pager.
+Also, you can override default pager via {} env. var.
+""".format(
+    DVC_PAGER
+)
+
+
+def find_pager():
+    if not sys.stdout.isatty():
+        return pydoc.plainpager
+
+    if is_exec_found(DEFAULT_PAGER):
+        pager_cmd = os.getenv(DVC_PAGER, DEFAULT_PAGER_FORMATTED)
+
+        def less_pager(text):
+            return pydoc.tempfilepager(pydoc.plain(text), pager_cmd)
+
+        return less_pager
+
+    logger.info(boxify(DVC_PAGER_INFO, border_color="yellow"))
```

Comment:
Btw, could you show a picture of how this info would look like? Btw, maybe this message is more suitable for a warning? E.g. info is directed to stdout and mixing it with the graph itself that is printed to stdout too is a bit strange.

Reply:
It's here
https://user-images.githubusercontent.com/298994/69914383-220f9680-1444-11ea-85da-de9a42683a85.png

in this thread
https://github.com/iterative/dvc/pull/2815/files#r352343840

by some reason the print screen is blue O_o. If you prefer non-blue, let me know.

I think it's ok, did this test
```
(_venv) [23:03:23] xliiv@alef /home/xliiv/workspace/dvc-project/example-pipeline (0) 
> dvc pipeline show --ascii run.dvc 1>1 2>2
(_venv) [23:03:35] xliiv@alef /home/xliiv/workspace/dvc-project/example-pipeline (0) 
> cat 1
+-------+******   +-------+****     +-------+         +-------+         +-------+         +-------+         +-------+         +-------+     ****+-------+    *****+--------+  
| 1.dvc |      ***| 2.dvc |    *****| 3.dvc |****     | 4.dvc |**       | 5.dvc |         | 6.dvc |       **| 7.dvc |     ****| 8.dvc |*****    | 9.dvc |****     | 10.dvc |  
+-------+         +-------+*********+-------+*********+-------+  *****  +-------+         +-------+   ****  +-------+*********+-------+*********+-------+         +--------+  
                                       *****************************  ****       *        *      *****   ******** *******************                                         
                                                   ****************************   **    **   ****************************                                                     
                                                               *******************  *  *  *******************                                                                 
                                                                           *****+---------+******                                                                             
                                                                                | run.dvc |                                                                                   
                                                                                +---------+                                                                                   
                                                                                              (_venv) [23:03:36] xliiv@alef /home/xliiv/workspace/dvc-project/example-pipeline (0) 
> cat 2

(_venv) [23:03:37] xliiv@alef /home/xliiv/workspace/dvc-project/example-pipeline (0)
> dvc pipeline show --ascii run.dvc
+-----------------------------------------------------------------------------------------------+
|                                                                                               |
|                              Less command line tool is missing.                               |
|     Install & add less tool to your PATH env. var. to automatically send output to pager.     |
|                 Also, you can override default pager via DVC_PAGER env. var.                  |
|                                                                                               |
|                                                                                               |
+-----------------------------------------------------------------------------------------------+

+-------+******   +-------+****     +-------+         +-------+         +-------+         +-------+         +-------+         +-------+     ****+-------+    *****+--------+
| 1.dvc |      ***| 2.dvc |    *****| 3.dvc |****     | 4.dvc |**       | 5.dvc |         | 6.dvc |       **| 7.dvc |     ****| 8.dvc |*****    | 9.dvc |****     | 10.dvc |
+-------+         +-------+*********+-------+*********+-------+  *****  +-------+         +-------+   ****  +-------+*********+-------+*********+-------+         +--------+
                                       *****************************  ****       *        *      *****   ******** *******************  
                                                   ****************************   **    **   ****************************              
                                                               *******************  *  *  *******************                          
                                                                           *****+---------+******                                      
                                                                                | run.dvc |                                            
                                                                                +---------+                                                                                   ```

Reply:
@xliiv The width is way over 80 chars for the box, which is way too much and gets awful when the lines overflow because of the box edges mixing with the text. With no box it would just overflow without real consequences. Feels like we could do better. Maybe something like
```
logger.warning(
    "Unable to find `less` in the PATH. Check out "
    "man.dvc.org/doc/command-reference/pipeline/show for more info."
)
```
where we could describe the situation in more details and provide a recipe on where to get `less` from on windows? CC @jorgeorpinel @casperdcl

Reply:
> get `less` on windows?

- maybe use recent version of powershell?
- install git-bash/cygwin/msys
- install gnuwin32-less
- install WSL
- give up and actually install linux

Reply:
makes sense.
so i'm going to use
```
logger.warning(
    "Unable to find `less` in the PATH. Check out "
    "man.dvc.org/doc/command-reference/pipeline/show for more info."
)```

Reply:
thanks and done. Also just opened this draft https://github.com/iterative/dvc.org/pull/831

Reply:
Agee with Ruslan. Please use this code though:

```py
logger.warning(
   "Unable to find `less` in the system PATH. Please refer to "
   "man.dvc.org/pipeline/show for more info."
)
```

> where in the docs could we describe the situation in more details and provide a recipe on where to get `less` from on windows?

Maybe in https://dvc.org/doc/user-guide/running-dvc-on-windows? Linked to from https://dvc.org/doc/command-reference/pipeline/show and other appropriate places of course. Please open an issue on dvc.org repo for this!

Reply:
@jorgeorpinel good idea, here's the task
https://github.com/iterative/dvc.org/issues/836